shunting-yard-replicator/src/main/java/com/hotels/shunting/yard/replicator/exec/messaging/MessageReaderAdapter.java

```diff
@@ -57,14 +63,26 @@ public void close() throws IOException {
 
   @Override
   public Optional<MetaStoreEvent> read() {
-    Optional<ListenerEvent> event = messageReader.read();
+    Optional<MessageEvent> event = messageReader.read();
     if (event.isPresent()) {
-      return Optional.of(map(event.get()));
+      MessageEvent messageEvent = event.get();
+      deleteMessage(messageEvent);
+      return Optional.of(map(messageEvent.getEvent()));
     } else {
       return Optional.empty();
     }
   }
 
+  private void deleteMessage(MessageEvent event) {
+    try {
+      String receiptHandle = event.getMessageProperties().get(SqsMessageProperty.SQS_MESSAGE_RECEIPT_HANDLE);
+      messageReader.delete(receiptHandle);
+      log.info("Message deleted successfully");
```

Comment:
should be log.debug otherwise this message will fill up the log file.

Reply:
SY doesn't take action on most of the messages but still deletes them from queue.

Reply:
Done, thanks